kratos/processes/set_initial_state_process.h

```diff
@@ -0,0 +1,191 @@
+    ///@{
+
+    /// Default constructor.
+    SetInitialStateProcess(ModelPart& rModelPart) :
+        mrModelPart(rModelPart)
+    {
+        const SizeType voigt_size = (TDim == 3) ? 6 : 3;
+
+        mInitialStrain.resize(voigt_size, false);
+        mInitialStress.resize(voigt_size, false);
+        mInitialF.resize(TDim, TDim, false);
+
+        noalias(mInitialStrain) = ZeroVector(voigt_size);
+        noalias(mInitialStress) = ZeroVector(voigt_size);
+        noalias(mInitialF)      = ZeroMatrix(TDim, TDim);
+    }
+
+    /// Full constructor.
+    SetInitialStateProcess(ModelPart& rModelPart,
+                           const Vector& rInitialStrain,
+                           const Vector& rInitialStress,
+                           const Matrix& rInitialF) :
+        mrModelPart(rModelPart), mInitialStrain(rInitialStrain), 
+        mInitialStress(rInitialStress), mInitialF(rInitialF)
+    {
+    }
+
+    /// Destructor.
+    ~SetInitialStateProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /// this function will be executed at every time step BEFORE performing the solve phase
+    void ExecuteInitializeSolutionStep() override
+    {
+        const auto it_elem_begin = mrModelPart.ElementsBegin();
+        const auto& integration_points = it_elem_begin->GetGeometry().IntegrationPoints(it_elem_begin->GetIntegrationMethod());
+
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(mrModelPart.Elements().size()); i++) {
+            auto it_elem = it_elem_begin + i;
+
+            Vector aux_initial_strain = mInitialStrain;
+            Vector aux_initial_stress = mInitialStrain;
+            Matrix aux_initial_F      = mInitialF;
+
+            // If the values are set element-wise have priority
+            if (it_elem->GetGeometry().Has(INITIAL_STRAIN_VECTOR)) {
+                noalias(aux_initial_strain) = (it_elem->GetGeometry()).GetValue(INITIAL_STRAIN_VECTOR);
+            }
+            if (it_elem->GetGeometry().Has(INITIAL_STRESS_VECTOR)) {
+                noalias(aux_initial_stress) = (it_elem->GetGeometry()).GetValue(INITIAL_STRESS_VECTOR);
+            }
+            if (it_elem->GetGeometry().Has(INITIAL_DEFORMATION_GRADIENT_MATRIX)) {
+                noalias(aux_initial_F) = (it_elem->GetGeometry()).GetValue(INITIAL_DEFORMATION_GRADIENT_MATRIX);
+            }
+            
+            InitialState initial_state = InitialState(aux_initial_strain, aux_initial_strain, aux_initial_F);
+
+            // Assign the values to the GP of the element
+
+            std::vector<ConstitutiveLaw::Pointer> constitutive_law_vector;
+            it_elem->CalculateOnIntegrationPoints(CONSTITUTIVE_LAW, constitutive_law_vector, mrModelPart.GetProcessInfo());
```

Comment:
I am not sure if this flexible enough. This relies on a spcefic way the constitutive laws are implemented in an element. Note that this is NOT enforced through the baseclass of `Element`, hence I think we should also avoid this implicit restriction here
why don't you use `SetValuesOnIntegrationPoints` instead and let the element handle this?

Reply:
Hi @philbucher ! In fact the method  `CalculateOnIntegrationPoints` can be found in the `element.h` :D

Reply:
but CONSTITUTIVE_LAW and especially the corresponding vectors are not!
Note that not all elements implement the Constitutive laws as a vector (e.g. the truss & the shells)

=> hence this is an implementation detail of the corresponding elements and should not be enforced

Reply:
The only thing that I dont really like is the fact of involving the elements in terms of Strains and Stresses... With the SetValuesOnIntegrationPoints I'll have to implement the interaction of the Element and the CL. 
I suppose that I'll have also to add a new variable like `INITIAL_STATE` in order to set everithing

Reply:
I think that @RiccardoRossi  has something to say about this! 👍

Reply:
I spoke with @RiccardoRossi and we agreed to move forward with your current proposal
For the solids the `CONSTITUTIVE_LAW` is straight forward to use, and for the other elements we can internally construct it

FYI @KlausBSautter also the membrane has a `std::vector<ConstitutiveLaw::Pointer> mConstitutiveLawVector` too so it will also be straight forward for you 

Overall I think it makes sense to use this for future elements too if possible/applicable